ui/components/app/add-network/add-network.js

```diff
@@ -1,168 +1,298 @@
-import React, { useContext } from 'react';
-import { useSelector } from 'react-redux';
-import PropTypes from 'prop-types';
+import React, { useContext, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
 import { I18nContext } from '../../../contexts/i18n';
 import Box from '../../ui/box';
 import Typography from '../../ui/typography';
 import {
   ALIGN_ITEMS,
-  BLOCK_SIZES,
   COLORS,
   DISPLAY,
   FLEX_DIRECTION,
   FONT_WEIGHT,
   TYPOGRAPHY,
   JUSTIFY_CONTENT,
+  SIZES,
 } from '../../../helpers/constants/design-system';
 import Button from '../../ui/button';
-import IconCaretLeft from '../../ui/icon/icon-caret-left';
 import Tooltip from '../../ui/tooltip';
 import IconWithFallback from '../../ui/icon-with-fallback';
 import IconBorder from '../../ui/icon-border';
-import { getTheme } from '../../../selectors';
+import {
+  getTheme,
+  getFrequentRpcListDetail,
+  getUnapprovedConfirmations,
+} from '../../../selectors';
 import { THEME_TYPE } from '../../../pages/settings/experimental-tab/experimental-tab.constant';
 
-const AddNetwork = ({
-  onBackClick,
-  onAddNetworkClick,
-  onAddNetworkManuallyClick,
-  featuredRPCS,
-}) => {
+import {
+  ENVIRONMENT_TYPE_FULLSCREEN,
+  ENVIRONMENT_TYPE_POPUP,
+  MESSAGE_TYPE,
+} from '../../../../shared/constants/app';
+import { requestUserApproval } from '../../../store/actions';
+import Popover from '../../ui/popover';
+import ConfirmationPage from '../../../pages/confirmation/confirmation';
+import { FEATURED_RPCS } from '../../../../shared/constants/network';
+import { ADD_NETWORK_ROUTE } from '../../../helpers/constants/routes';
+import { getEnvironmentType } from '../../../../app/scripts/lib/util';
+
+const AddNetwork = () => {
   const t = useContext(I18nContext);
-  const theme = useSelector(getTheme);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const frequentRpcList = useSelector(getFrequentRpcListDetail);
+
+  const frequentRpcListChainIds = Object.values(frequentRpcList).map(
+    (net) => net.chainId,
+  );
 
   const infuraRegex = /infura.io/u;
 
-  const nets = featuredRPCS
-    .sort((a, b) => (a.ticker > b.ticker ? 1 : -1))
-    .slice(0, 8);
+  const nets = FEATURED_RPCS.sort((a, b) =>
+    a.ticker > b.ticker ? 1 : -1,
+  ).slice(0, FEATURED_RPCS.length);
+
+  const notFrequentRpcNetworks = nets.filter(
+    (net) => frequentRpcListChainIds.indexOf(net.chainId) === -1,
+  );
+  const unapprovedConfirmations = useSelector(getUnapprovedConfirmations);
+  const [showPopover, setShowPopover] = useState(false);
+  const [theme, setTheme] = useState(useSelector(getTheme));
+
+  if (theme === THEME_TYPE.OS) {
+    const osTheme = window?.matchMedia('(prefers-color-scheme: dark)')?.matches
+      ? THEME_TYPE.DARK
+      : THEME_TYPE.LIGHT;
+
+    setTheme(osTheme);
+  }
```

Comment:
This theme stuff seems out of place -- does it need to be here?

Reply:
Yes, we need it here. If a theme is `os` then we have to find out which one to use (light or dark). That is being checked here.

Reply:
@darkwing This is to pass the correct theme to a tooltip component below. Is there something different that should happen if `theme === THEME_TYPE.OS`? How is that case handled in the rest of the app?

Reply:
OK, I see that now; I'm wondering if we should move this theme check to the actual Tooltip component itself?  I'm also wondering if there's any advantage to setting the theme in local state.

Reply:
Created a PR to fix this https://github.com/MetaMask/metamask-extension/pull/14992
Theme logic shouldn't be required once this has been merged to develop